CMakeLists.txt

```diff
@@ -178,11 +176,11 @@ endif()
 set(DLL_PATH_LIST
     "${DLL_PATH_LIST};PATH=path_list_append:${LIBHWLOC_LIBRARY_DIRS}/../bin")
 
-if(UMF_BUILD_LIBUMF_POOL_SCALABLE)
-    pkg_check_modules(TBB tbb)
-    if(NOT TBB_FOUND)
-        find_package(TBB REQUIRED tbb)
-    endif()
+pkg_check_modules(TBB tbb)
```

Comment:
It means that now we have strict build-time dependency on the TBB library.

The goal of this PR is to make scalable_pool implementation to be part of libumf (not a separate binary) because scalable pool implementation does not require linking libumf with tbbmalloc (tbb malloc is loaded via dlopen at runtime, the same story as with L0 provider).

The next question is whether we want strict build-time dependency to TBB? As I can see `pool_scalable.c` includes the `#include <tbb/scalable_allocator.h>` it is required only for one symbol: `TBBMALLOC_OK`. If we can avoid usage of this symbol we do not need the build time dependency to TBB. Otherwise, if build-time dependency to TBB is needed we should consider the same way like with L0 provider:

- keep the build option for scalable pool
- for the oneAPI package we will build UMF with scalable pool, but by default it is disabled to simplify life for developers who have no TBB installed on their development machine.

I vote for the removing build time dependency to the TBB because it should be simple.

Reply:
I removed `#include <tbb/scalable_allocator.h>` from the code and make TBB an "optional" component. The only reason I kept this fragment of code is for "set(DLL_PATH_LIST ..." for Windows that makes testing easier.